server/src/main.rs

```diff
@@ -1,17 +1,55 @@
+use clap::{crate_version, App, SubCommand};
 use stackable_operator::crd::CustomResourceExt;
-use stackable_operator::{client, error};
+use stackable_operator::{cli, client};
 use stackable_spark_crd::SparkCluster;
 use stackable_spark_crd::{Restart, Start, Stop};
 use tracing::{error, info};
 
 #[tokio::main]
-async fn main() -> Result<(), error::Error> {
+async fn main() -> anyhow::Result<()> {
     stackable_operator::logging::initialize_logging("SPARK_OPERATOR_LOG");
 
     info!("Starting Stackable Operator for Apache Spark");
 
+    // Handle CLI arguments
+    let matches = App::new("Spark Operator")
+        .author("Stackable GmbH - info@stackable.de")
+        .about("Stackable Operator for Apache Spark")
+        .version(crate_version!())
+        .arg(cli::generate_productconfig_arg())
+        .subcommand(
+            SubCommand::with_name("crd")
+                .subcommand(cli::generate_crd_subcommand::<SparkCluster>())
+                .subcommand(cli::generate_crd_subcommand::<Restart>())
+                .subcommand(cli::generate_crd_subcommand::<Start>())
+                .subcommand(cli::generate_crd_subcommand::<Stop>()),
+        )
+        .get_matches();
+
+    if let ("crd", Some(subcommand)) = matches.subcommand() {
+        if cli::handle_crd_subcommand::<SparkCluster>(subcommand)? {
+            return Ok(());
+        };
+        if cli::handle_crd_subcommand::<Restart>(subcommand)? {
+            return Ok(());
+        };
+        if cli::handle_crd_subcommand::<Start>(subcommand)? {
+            return Ok(());
+        };
+        if cli::handle_crd_subcommand::<Stop>(subcommand)? {
+            return Ok(());
+        };
+    }
+
+    let paths = vec![
```

Comment:
Shouldn't we maybe move these to the common crate as constants?

Reply:
Yes, at some point :)
They are a bit dynamic because they include the operator specific directory but the answer is still "yes"